utils/message_html_to_component.jsx

```diff
@@ -109,6 +111,7 @@ export function messageHtmlToComponent(html, isRHS, options = {}) {
         processingInstructions.push({
             shouldProcessNode: (node) => node.type === 'tag' && node.name === 'img',
             processNode: (node) => {
```

Comment:
I'm slightly worried about binding the underlying storage bit to the index being computed here. If someone goes and edits their post, the index will change, but not the storage, so users might see their configurations flip.

Would it make sense to instead use the `src` attribute as the key here? This has the downside that duplicates of /same/ src posted in the same post will get collapsed together, but that seems unlikely. Plus if the post is edited to show a /new/ inline image, the behaviour will correctly reset to the defaults for the new image.

Reply:
I have always been tentative about this. I was simply trying to assess which behavior seems more "unacceptable" or confusing from a user perspective - either two images sharing the same collapse/expand state (which will happen in case of using src as a key) or shuffling this state completely in case of image editing (when we use indexes instead).

I think I agree with you that index-based approach causes more "harm" in terms of usability. Because post editing will happen for sure, whereas multiple collapseable inline images with the same URL don't seem to appear that likely. 

I will fix this shortly.

Reply:
Fixed.

Reply:
That's a good point about editing posts making it weird. The problem with just using the URL is that having the same image in a post twice will cause both of them to expand/collapse together. That said, that's probably a less common case than editing a post with a URL, so I'd think it's better to have work this way even if there is a weird case.

Ideally, I guess out best bet would be to add an index for each time an image URL occurs in a post, so that the first and second instance of each image would be tracked separately. That may still be weird if the user heavily edits a post and, for example, removes one instance of the image, but I think that's probably the best we could do since you're otherwise having to spend a lot more time trying to figure out whether or not the user meant to remove the first or second image which is obviously very hard.

That all said, I think the per-URL method probably makes the most sense for now since it'll cover 99% of cases. I think the case I mentioned is enough of an edge case that we can accept it as a known issue to fix if someone comes up with a more practical use case when that might occur.